### DISCUSSION TITLE
busboy and uws.getParts

### DISCUSSION THREAD

**Discussion by joshxyzhimself** [kartikk221/hyper-express#120] [Q&A] [answered]
Hi just curious, is there a reason why busboy is better than uws.getParts for parsing multipart body?

- https://github.com/kartikk221/hyper-express/blob/master/src/components/http/Request.js
- https://unetworking.github.io/uWebSockets.js/generated/index.html#getParts

**Comment by kartikk221** [comment, marked as answer]
Hi @joshxyzhimself The `uWS.getParts()` method requires the full body as a `RecognizedString` which would mean that the incoming body has to be fully available in memory for uWS to output the Multipart fields, this is simply inefficient and would in some cases exhaust the memory where large files are being uploaded.

Busboy on the other hand acts as an asynchronous parser which gradually consumes the incoming stream of data and emits events to allow for consumption of multipart fields as they are detected through the body stream. Furthermore, HyperExpress will automatically pause/resume the incoming body stream in-between Multipart events to allow for the user to perform some async processing thus giving the developer most control while being the most memory efficient as data is gradually streamed and only kept If it is being used.
